Improvements in the func calls and minor changes.

abhijitherekar opened the pull request:
1. Fixed lot of linting issues.
2. improve the file count func.
3. make changes to the matches to accept the file and msg index.

mheese commented on .speakeasy/gen.yaml:
```diff
@@ -16,7 +16,7 @@ generation:
     oAuth2PasswordEnabled: true
 python:
-  version: 0.5.2
+  version: 0.5.3
```
please use this PR as a chance to remove the `rich` package again

satyamsi commented on src/acuvity/apexextend.py:
```diff
@@ -183,7 +183,7 @@ async def scan_async(
             raise ValueError("Cannot process the guard config") from e
 
-        return ScanResponseMatch(raw_scan_response, gconfig, *messages, files=files)
+        return ScanResponseMatch(raw_scan_response, gconfig, files=files)
```
i still think we only should be passing num_files and not files.

mheese replied:
why do you say that?

abhijitherekar replied:
either somewhere we need to count for the files. 

Its better because the logic is tied to this class.

Else, we have to do in the apex-extend class.

mheese replied:
Took me some time to understand why we are doing it like this. The problem really is that the `ScanResponseMatch` class is essentially not simply tracking the request at all. So whatever is being done in that class right now all feels like workarounds.

We discussed this now a lot of times already: unfortunately the helpers cannot only look at the scan response, but also need to look at the request. Originally that was because it needs to build request parameters, now here this shows that the display of specific matches is affected. `files` is special in the SDK because they don't match anything in the request as the SDK is reading the file and building the request for it. It would probably help to reallly track everything request related in the ScanResponseMatch, then there are no shenanigans needed.

abhijitherekar replied:
yeah, completely agree. Earlier we didn't really want the request but as we are adding things here, it makes sense to have the request in the scanresponsematch and use everything from there. In future also, if we need anything than we can use it.

mheese commented on src/acuvity/response/helper.py:
```diff
@@ -154,18 +160,18 @@ def _get_text_detections(
         guard: Guard,
         detection_type: TextualdetectionType,
-        detections: List[Textualdetection] | None,
+        detections: Union[List[Textualdetection], None],
```
these are good changes because this would actually break on Python 3.9

When you test/develop against these SDKs, please use the minimum supported version when developing. This way you catch problems like this right away.

mheese reviewed (changes requested):
now that speakeasy is running again everything, this fails actually with something which is probably related because of the added dependencies:

```

» Compile SDK...

INFO    Running command: poetry version 0.5.3   {"path":"/Users/marcusheese/src/acuvity-python"}
INFO    Running command: poetry lock    {"path":"/Users/marcusheese/src/acuvity-python"}
INFO    Running command: poetry run python scripts/prepare-readme.py    {"path":"/Users/marcusheese/src/acuvity-python"}
INFO    Running command: poetry install --with=dev      {"path":"/Users/marcusheese/src/acuvity-python"}
INFO    Running command: poetry run python -m compileall -q .   {"path":"/Users/marcusheese/src/acuvity-python"}
INFO    Running command: poetry run pylint src/acuvity  {"path":"/Users/marcusheese/src/acuvity-python"}
INFO    Running command: poetry run mypy src/acuvity    {"path":"/Users/marcusheese/src/acuvity-python"}
WARN    failed running commands:
src/acuvity/guard/config.py:7: error: Library stubs not installed for "yaml"  [import-untyped]
src/acuvity/guard/config.py:7: note: Hint: "python3 -m pip install types-PyYAML"
src/acuvity/guard/config.py:7: note: (or run "mypy --install-types" to install all missing stub packages)
src/acuvity/guard/config.py:7: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 60 source files)


exit status 1

Step Failed: Compile SDK


Step Failed: Workflow


Step Failed: Target: python


Step Failed: Generating Python SDK


──────────────────────────────────────────────────────────── ↑ Workflow run logs ↑ ────────────────────────────────────────────────────────────
failed to generate SDKs for python ✖
Failed to run with Speakeasy version 1.469.1: failed to run with version 1.469.1: exit status 1

failed to run with version 1.469.1: exit status 1
```

let's please get that to a good state + make sure we're using python 3.8 typing in all files.

abhijitherekar commented:
> now that speakeasy is running again everything, this fails actually with something which is probably related because of the added dependencies:
> 
> ```
> 
> » Compile SDK...
> 
> INFO    Running command: poetry version 0.5.3   {"path":"/Users/marcusheese/src/acuvity-python"}
> INFO    Running command: poetry lock    {"path":"/Users/marcusheese/src/acuvity-python"}
> INFO    Running command: poetry run python scripts/prepare-readme.py    {"path":"/Users/marcusheese/src/acuvity-python"}
> INFO    Running command: poetry install --with=dev      {"path":"/Users/marcusheese/src/acuvity-python"}
> INFO    Running command: poetry run python -m compileall -q .   {"path":"/Users/marcusheese/src/acuvity-python"}
> INFO    Running command: poetry run pylint src/acuvity  {"path":"/Users/marcusheese/src/acuvity-python"}
> INFO    Running command: poetry run mypy src/acuvity    {"path":"/Users/marcusheese/src/acuvity-python"}
> WARN    failed running commands:
> src/acuvity/guard/config.py:7: error: Library stubs not installed for "yaml"  [import-untyped]
> src/acuvity/guard/config.py:7: note: Hint: "python3 -m pip install types-PyYAML"
> src/acuvity/guard/config.py:7: note: (or run "mypy --install-types" to install all missing stub packages)
> src/acuvity/guard/config.py:7: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
> Found 1 error in 1 file (checked 60 source files)
> 
> 
> exit status 1
> 
> Step Failed: Compile SDK
> 
> 
> Step Failed: Workflow
> 
> 
> Step Failed: Target: python
> 
> 
> Step Failed: Generating Python SDK
> 
> 
> ──────────────────────────────────────────────────────────── ↑ Workflow run logs ↑ ────────────────────────────────────────────────────────────
> failed to generate SDKs for python ✖
> Failed to run with Speakeasy version 1.469.1: failed to run with version 1.469.1: exit status 1
> 
> failed to run with version 1.469.1: exit status 1
> ```
> 
> let's please get that to a good state + make sure we're using python 3.8 typing in all files.

I am not sure what I have do here, it says to run... python3 -m pip install types-PyYAML

as we added yaml in the code.

mheese commented on .speakeasy/gen.yaml:
```diff
@@ -16,13 +16,13 @@ generation:
       PyYAML: ^6.0.2
       pyjwt: ^2.9.0
-      rich: ^13.9.4
+      types-PyYAML: ^6.0.12
```
this is needed as a `dev` dependency only I would think.

abhijitherekar replied:
yeah, looks like that. will add it there.

mheese commented on examples/basic_file.py:
```diff
@@ -14,7 +14,7 @@
 print("--------------------------------------------------------------------------------")
 print("Scenario: single file with prompt injection detection")
-file="./test_data/pi-test.txt"
+file="./examples/test_data/pi-test.txt"
```
Let's be a bit smarter here and let's make the file references relative to the file itself. It's not really good practice when there are expectations on the working directory.

abhijitherekar replied:
hmm, not sure why its showing as part of my PR. This were Satyam's changes, I thought it was merged maybe he added it here. 

Will me take care of it.

mheese commented on examples/run.sh:
```diff
@@ -1,22 +1,22 @@
 
 echo "Running basic_file.py ..."
-python basic_file.py
+python examples/basic_file.py
```
same here in the bash script. add the good old

```
SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```

and make use of that

abhijitherekar replied:
same as the previous comment but will take care of it.

mheese reviewed (changes requested):
This is an improvement to what was there before, so this should get merged soon.

The main thing really that irks me at the moment is that the `ScanResponseMatch` isn't fully tracking everything request related which leads to hacks for file counting etc. I'm also not entirely sure why we removed `messages` now from the workarounds.

Also, the `types-PyYAML` package should get added as a dev dependency. That'll for sure fix the speakeasy problem and it'll make it disappear from our dependencies.

abhijitherekar commented:
> This is an improvement to what was there before, so this should get merged soon.
> 
> The main thing really that irks me at the moment is that the `ScanResponseMatch` isn't fully tracking everything request related which leads to hacks for file counting etc. I'm also not entirely sure why we removed `messages` now from the workarounds.
> 
> Also, the `types-PyYAML` package should get added as a dev dependency. That'll for sure fix the speakeasy problem and it'll make it disappear from our dependencies.

I have addressed the issues, for the request as part of the scanresponse, I would want it be part of another PR, as this has already lot of fixes which are necessary.

mheese reviewed (approved):
lgtm
